device/transport/amqp/src/amqp.ts

```diff
@@ -372,20 +376,27 @@ export class Amqp extends EventEmitter implements DeviceTransport {
 
             /*Codes_SRS_NODE_DEVICE_AMQP_16_025: [The `sendEvent` method shall create and attach the d2c link if necessary.]*/
             /*Codes_SRS_NODE_DEVICE_AMQP_18_006: [The `sendOutputEvent` method shall create and attach the d2c link if necessary.]*/
-            if (!this._d2cLink) {
-              this._amqp.attachSenderLink(this._d2cEndpoint, null, (err, link) => {
-                if (err) {
-                  handleResult('AMQP Transport: Could not send', sendCallback)(err);
-                } else {
-                  debug('got a new D2C link');
-                  this._d2cLink = link;
-                  this._d2cLink.on('error', this._d2cErrorListener);
-                  this._d2cLink.send(amqpMessage, handleResult('AMQP Transport: Could not send', sendCallback));
-                }
-              });
-            } else {
+            if (this._d2cLink) {
```

Comment:
You're going to have to add unit tests simply to keep our coverage numbers up AND, this should have some tests anyway.